app/models/services/crossref.rb

```diff
@@ -0,0 +1,17 @@
+class Crossref
+  def initialize(options={})
+    @base_url = 'http://api.crossref.org/'
+    @metadata_url = @base_url + "works/"
+    @funder_url = @base_url + "funders/"
+  end
+
+  def get_doi_metadata(doi)
+    metadata_uri = URI.parse(@metadata_url+doi)
+    begin
+      JSON.parse Net::HTTP.get(metadata_uri)
+    rescue JSON::ParserError => e
+      return nil
+    end
```

Comment:
The rescue block handles doi's that aren't found in crossref.  [Basically the api returns this if it can't find the doi:](http://api.crossref.org/works/10.1037/poop)

For this method, it's meant to grab the data if it exists.  The JSON parse breaks if the doi is not found because the API doesn't return JSON.  So I have this method return a nil and  I handle that nil in each #find_paper for the locators:


```
    crossref = Crossref.new
    if (result = crossref.get_doi_metadata(self.locator_id))
      data = result['message']
      paper = self.import_data_to_paper(paper,data,'crossref')
    end
```

The reason I have it continue is so I can search pubmed for that doi as well.  Crossref's API only has doi's listed by crossref, which should be most, but there's a chance that the doi will exist in pubmed and not crossref.

Alternatively, I can handle the JSON parse outside this method if we want to remove this error block from this method.  I was trying to follow the convention of "return an object or nil" but let me know if there's a better way.